src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -47,26 +130,707 @@ protected override X509Certificate2 SaveCertificateCore(X509Certificate2 certifi
+        if (Directory.Exists(firefoxSnapDir))
+        {
+            var profileDirs = GetFirefoxProfiles(firefoxSnapDir);
+            foreach (var profileDir in profileDirs)
+            {
+                nssDbs.Add(new NssDb(profileDir, isFirefox: true));
+            }
+        }
+
+        return nssDbs;
+    }
+
+    [GeneratedRegex("OPENSSLDIR:\\s*\"([^\"]+)\"")]
+    private static partial Regex OpenSslVersionRegex();
+
+    /// <remarks>
+    /// It is the caller's responsibility to ensure that <see cref="OpenSslCommand"/> is available.
+    /// </remarks>
+    private static bool TryGetOpenSslDirectory([NotNullWhen(true)] out string? openSslDir)
+    {
+        openSslDir = null;
+
+        try
+        {
+            var processInfo = new ProcessStartInfo(OpenSslCommand, $"version -d")
+            {
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            using var process = Process.Start(processInfo);
+            var stdout = process!.StandardOutput.ReadToEnd();
+
+            process.WaitForExit();
+            if (process.ExitCode != 0)
+            {
+                Log.UnixOpenSslVersionFailed();
+                return false;
+            }
+
+            var match = OpenSslVersionRegex().Match(stdout);
+            if (!match.Success)
+            {
+                Log.UnixOpenSslVersionParsingFailed();
+                return false;
+            }
+
+            openSslDir = match.Groups[1].Value;
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Log.UnixOpenSslVersionException(ex.Message);
+            return false;
+        }
+    }
+
+    /// <remarks>
+    /// It is the caller's responsibility to ensure that <see cref="OpenSslCommand"/> is available.
+    /// </remarks>
+    private static bool TryGetOpenSslHash(string certificatePath, [NotNullWhen(true)] out string? hash)
+    {
+        hash = null;
+
+        try
+        {
+            // c_rehash actually does this twice: once with -subject_hash (equivalent to -hash) and again
+            // with -subject_hash_old.  Old hashes are only  needed for pre-1.0.0, so we skip that.
+            var processInfo = new ProcessStartInfo(OpenSslCommand, $"x509 -hash -noout -in {certificatePath}")
+            {
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            using var process = Process.Start(processInfo);
+            var stdout = process!.StandardOutput.ReadToEnd();
+
+            process.WaitForExit();
+            if (process.ExitCode != 0)
+            {
+                Log.UnixOpenSslHashFailed(certificatePath);
+                return false;
+            }
+
+            hash = stdout.Trim();
+            return true;
+        }
+        catch (Exception ex)
+        {
+            Log.UnixOpenSslHashException(certificatePath, ex.Message);
+            return false;
+        }
+    }
```

Comment:
I'm a bit confused here, why do we need to compute the hash using OpenSSL, isn't our `c_rehash` replacement enough?

Reply:
This is called from our `c_rehash` replacement.  Like the original tool, we delegate the actual hashing to openssl.